Source/CombatExtended/CombatExtended/StatWorkers/StatWorker_ArmorPartial.cs

```diff
@@ -19,30 +19,41 @@ public override string GetExplanationFinalizePart(StatRequest req, ToStringNumbe
                 {
                     if (req.Thing.def.HasModExtension<PartialArmorExt>())
                     {
-                        var result = "CE_StatWorker_ArmorGeneral".Translate() + finalVal.ToString() + " \n \n" + "CE_StatWorker_ArmorSpecific".Translate();
+                        var result = "";
+
+                        if ((this.stat?.parts ?? null) != null)
+                        {
+                            foreach (var part in this.stat.parts)
+                            {
+                                result += part.ExplanationPart(req) + "\n";
+                            }
+                        }
+                        result += "\n" + "CE_StatWorker_ArmorGeneral".Translate() + finalVal.ToString() + " \n \n" + "CE_StatWorker_ArmorSpecific".Translate();
+
                         var ext = req.Thing.def.GetModExtension<PartialArmorExt>();
                         ;
                         foreach (ApparelPartialStat partstat in ext.stats)
                         {
-
-
-                            var value = 0f;
-                            if (req.Thing is Apparel)
+                            if (partstat.stat == this.stat)

```

Comment:
I tend to prefer early return/continue when possible, instead of wrapping nested layers of `if`.
Something like
```cs
if (parstat.stat != this.stat) { continue; }
// rest of code
```

Your call